File: scripts/download-logos.js
/**
 * download-logos.js
 * Run once:  node scripts/download-logos.js
 *
 * Downloads every tool logo from Clearbit and saves it to
 * /public/logos/<domain>.png  (dots in domain become dashes in filename)
 *
 * After this runs, the ToolCard component will automatically serve
 * the local files instead of making Clearbit requests on every page load.
 */

const https = require('https')
const fs    = require('fs')
const path  = require('path')

const OUT_DIR = path.join(__dirname, '..', 'public', 'logos')

const tools = [
  'granola.ai',
  'notebooklm.google.com',
  'notion.so',
  'raycast.com',
  'tldraw.com',
  'excalidraw.com',
  'mermaidchart.com',
  'mobbin.com',
  'refero.design',
  'banani.co',
  'superdesign.dev',
  'paper.design',
  'replit.com',
  'usegalileo.ai',
  'cursor.com',
  'claude.ai',
  'openai.com',
  'github.com',
  'vercel.com',
  'n8n.io',
  'zapier.com',
  'make.com',
  'airtable.com',
  'playwright.dev',
  'agentation.com',
  'dialkit.com',
  'scribehow.com',
  'krea.ai',
  'midjourney.com',
  'runwayml.com',
  'elevenlabs.io',
  'rive.app',
  'lottiefiles.com',
  'jitter.video',
  'gamma.app',
  'tome.app',
  'pitch.com',
  'canva.com',
]

if (!fs.existsSync(OUT_DIR)) fs.mkdirSync(OUT_DIR, { recursive: true })

const domainToFilename = (domain) => domain.replace(/\./g, '-') + '.png'

let done = 0

function download(domain) {
  const url      = `https://logo.clearbit.com/${domain}?size=128`
  const filename = domainToFilename(domain)
  const dest     = path.join(OUT_DIR, filename)

  const file = fs.createWriteStream(dest)
  https.get(url, (res) => {
    if (res.statusCode === 200) {
      res.pipe(file)
      file.on('finish', () => {
        file.close()
        console.log(`✓  ${domain}  →  public/logos/${filename}`)
        if (++done === tools.length) console.log('\nAll logos downloaded.')
      })
    } else {
      fs.unlink(dest, () => {})
      console.warn(`✗  ${domain}  (status ${res.statusCode} — fallback to initials)`)
      if (++done === tools.length) console.log('\nDone.')
    }
  }).on('error', (err) => {
    fs.unlink(dest, () => {})
    console.warn(`✗  ${domain}  (${err.message})`)
    if (++done === tools.length) console.log('\nDone.')
  })
}

console.log(`Downloading ${tools.length} logos to public/logos/ ...\n`)
tools.forEach(download)

File: download_logos.py
#!/usr/bin/env python3
"""Download logos from Clearbit and save to public/logos/"""

import os
import urllib.request
import urllib.error

# Target directory
logos_dir = os.path.join(os.path.dirname(__file__), "public", "logos")
os.makedirs(logos_dir, exist_ok=True)

domains = [
    "granola.ai",
    "notebooklm.google.com",
    "notion.so",
    "raycast.com",
    "tldraw.com",
    "excalidraw.com",
    "mermaidchart.com",
    "mobbin.com",
    "refero.design",
    "banani.co",
    "superdesign.dev",
    "paper.design",
    "replit.com",
    "usegalileo.ai",
    "cursor.com",
    "claude.ai",
    "openai.com",
    "github.com",
    "vercel.com",
    "n8n.io",
    "zapier.com",
    "make.com",
    "airtable.com",
    "playwright.dev",
    "agentation.com",
    "dialkit.com",
    "scribehow.com",
    "krea.ai",
    "midjourney.com",
    "runwayml.com",
    "elevenlabs.io",
    "rive.app",
    "lottiefiles.com",
    "jitter.video",
    "gamma.app",
    "tome.app",
    "pitch.com",
    "canva.com",
]

success = []
failed = []

for domain in domains:
    filename = domain.replace(".", "-") + ".png"
    filepath = os.path.join(logos_dir, filename)
    url = f"https://logo.clearbit.com/{domain}?size=128"

    try:
        req = urllib.request.Request(url, headers={"User-Agent": "Mozilla/5.0"})
        with urllib.request.urlopen(req, timeout=15) as response:
            content_type = response.headers.get("Content-Type", "")
            data = response.read()
            if "image" in content_type and len(data) > 100:
                with open(filepath, "wb") as f:
                    f.write(data)
                success.append(domain)
                print(f"OK:   {domain} -> {filename}")
            else:
                failed.append((domain, f"unexpected content-type: {content_type}"))
                print(f"FAIL: {domain} (content-type: {content_type})")
    except urllib.error.HTTPError as e:
        failed.append((domain, f"HTTP {e.code}"))
        print(f"FAIL: {domain} (HTTP {e.code})")
    except Exception as e:
        failed.append((domain, str(e)))
        print(f"FAIL: {domain} ({e})")

print(f"\n=== DONE: {len(success)} succeeded, {len(failed)} failed ===")
if failed:
    print("\nFailed:")
    for domain, reason in failed:
        print(f"  - {domain}: {reason}")
